sdcm/cluster_aws.py

```diff
@@ -593,6 +635,19 @@ def _instance_wait_safe(self, instance_method: Callable[P, R], *args: P.args, **
     def _wait_public_ip(self):
         ec2_instance_wait_public_ip(self._instance)
 
+    @cached_property
+    def cql_address(self):
+        # TODO: is it correct ???
```

Comment:
seems not correct:
self.scylla_network_configuration.broadcast_rpc_address should reflect what is configured, so when some part of code wants this address, should get proper one (need to have cql_address to be the same as broadcast_rpc_address). SCT should apply change to `scylla_network_config` to reflect that.

Reply:
I really don't remember why we have this here this one should be tied to `test_communication` 
and if it is meaning we are misusing this one, I'm guessing in one of the artifacts test,s we need to documented the exact reason if we have such

Reply:
if it's for cases we are running the test locally, it's o.k. to leave it here, but document what is the exact use case it's relevant for

Reply:
i.e. test run locally why cluster is in AWS with private addresses,  we need to consider if we support it or not.
so far we didn't, and you have to run with public addresses when running from local machine

Reply:
This address is mainly used by stress. It means - connect to the DB outside of the cluster. So if the Scylla allows to connect with private, we may fail to rich the node if the loader is not located in the same region. Like Israel wrote. So we still want to use public address.
Looks like

Reply:
is it possible to set broadcast address to private IP and use public IP for cql?
If not, broadcast_address should equal cql_address

Reply:
@juliayakovlev 

I think @soyacz is right, we just need to validate the running from local machine case,
and the azure artifact, that run on AWS with the db cluster in Azure  

if both of those are working, we don't need any more logic here

Reply:
> @juliayakovlev
> 
> I think @soyacz is right, we just need to validate the running from local machine case, and the azure artifact, that run on AWS with the db cluster in Azure
> 
> if both of those are working, we don't need any more logic here

I tried that (this is the reason why I put the question there) - did not work